### PR TITLE
Adding example of manually setting up NGINX Ingress

### PR DESCRIPTION
**Pull request by jasoonn**
## Why are these changes needed?

This example provides users a starting point to customize their own ingress for ray clusters. [ray-cluster.separate-ingress.yaml](https://github.com/ray-project/kuberay/blob/9765abb0dc2b48074f4fdb98443749617fdbe373/ray-operator/config/samples/ray-cluster.separate-ingress.yaml) provides a template for setting up a ray cluster and the corresponding ingress, which clearly shows the relationship between ray cluster and ingress.
### Information helpful for modifying the ingress in [ray-cluster.separate-ingress.yaml](https://github.com/ray-project/kuberay/blob/9765abb0dc2b48074f4fdb98443749617fdbe373/ray-operator/config/samples/ray-cluster.separate-ingress.yaml) 
* `spec.rules.http.paths.backend.service.name` should be the same as head service
* `spec.rules.http.paths.backend.service.port.number` should match the port specified for dashboard in the RayCluster
* `metadata.annotations.nginx.ingress.kubernetes.io/rewrite-target: /$1`
   Any characters captured by (.*) will be assigned to the placeholder $1, which is then used as a parameter in the rewrite-target annotation([rewrite-target doc](https://github.com/kubernetes/ingress-nginx/blob/main/docs/examples/rewrite/README.md#rewrite-target))
    For example, the ingress definition above will result in the following rewrites:
    * `$IP/$CLUSTER_NAME/` rewrites to `$IP/` ($1 maps to nothing)
    * `$IP/$CLUSTER_NAME/#/actors` rewrites to `$IP/#/actors` ($1 maps to #/actors)
    * `$IP/$CLUSTER_NAME/#/node` rewrites to `$IP/#/node` ($1 maps to #/node)
## Related issue number
#646 

## Demo 
* Step 1: Create kind cluster, for more reference can see in [here](https://kind.sigs.k8s.io/docs/user/ingress/).
image

* Step 2: Install NGINX ingress controller
image

* Step 3: Install KubeRay operator
image

* Step 4: Install RayCluster and create an ingress separately.
image

* Step 5: Check ingress created by Step 4.
image

* Step 6: Check `<ip>/raycluster-ingress/` on the browser.
image

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [x] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: docs/guidance/ingress.md [hunk 1]
```diff
@@ -3,6 +3,9 @@
 * KubeRay built-in ingress support: KubeRay will help users create a Kubernetes ingress when `spec.headGroupSpec.enableIngress: true`. Currently, the built-in support only supports simple NGINX setups. Note that users still need to install ingress controller by themselves. **We do not recommend using built-in ingress support in a production environment with complex routing requirements.**
   * [Example: NGINX Ingress on KinD (built-in ingress support)](#example-nginx-ingress-on-kind-built-in-ingress-support)
 
+* Manually setting up an ingress with NGINX:
+  * [Example: Manually setting up NGINX Ingress on KinD](#example-manually-setting-up-nginx-ingress-on-kind)
```
Comment (kevin85421) [inline docs/guidance/ingress.md:7]: Move this line under "Example: AWS Application Load Balancer (ALB) ...".

File: docs/guidance/ingress.md [hunk 2]
```diff
@@ -70,6 +73,61 @@ kubectl describe ingress raycluster-ingress-head-ingress
...
+  --for=condition=ready pod \
+  --selector=app.kubernetes.io/component=controller \
+  --timeout=90s
+
+# Step 3: Install KubeRay operator
+pushd helm-chart/kuberay-operator
+helm install kuberay-operator .
+
+# Step 4: Install RayCluster and create an ingress separately. 
+popd
```
Comment (kevin85421) [inline docs/guidance/ingress.md:111]: Please teach users how to modify the YAML file both in this doc and `ray-cluster.separate-ingress.yaml`. For example, "...backend.service.name" should be the same as head service.

File: docs/guidance/ingress.md [hunk 3]
```diff
@@ -70,6 +73,61 @@ kubectl describe ingress raycluster-ingress-head-ingress
...
+# ----        ----  --------
+# *
+#             /raycluster-ingress/(.*)   raycluster-ingress-head-svc:8265 (10.244.0.11:8265)
+# Annotations:  nginx.ingress.kubernetes.io/rewrite-target: /$1
+
+# Step 6: Check `<ip>/raycluster-ingress/` on your browser. You will see the Ray Dashboard.
+#        [Note] The forward slash at the end of the address is necessary. `<ip>/raycluster-ingress`
+#               will report "404 Not Found".
+```
+
```
Comment (kevin85421) [inline docs/guidance/ingress.md:130]: Add the link (https://kind.sigs.k8s.io/docs/user/ingress/) in both the PR description and `ingress.md` to help users resolve issues by themselves.

**Comment by jasoonn** [comment]
@kevin85421 Thanks for your suggestion. I have updated my pr.

File: docs/guidance/ingress.md [hunk 4]
```diff
@@ -140,3 +142,62 @@ kubectl describe ingress ray-cluster-ingress
...
+  --selector=app.kubernetes.io/component=controller \
+  --timeout=90s
+
+# Step 3: Install KubeRay operator
+pushd helm-chart/kuberay-operator
+helm install kuberay-operator .
+popd
+
+# Step 4: Install RayCluster and create an ingress separately. 
+# If you want to change the setting of ingress, you can edit the ingress portion in
```
Comment (DmitriGekhtman) [inline docs/guidance/ingress.md:182]:
```suggestion
# If you want to change ingress settings, you can edit the ingress portion in 
```

File: docs/guidance/ingress.md [hunk 5]
```diff
@@ -140,3 +142,62 @@ kubectl describe ingress ray-cluster-ingress
...
+popd
+
+# Step 4: Install RayCluster and create an ingress separately. 
+# If you want to change the setting of ingress, you can edit the ingress portion in 
+# `ray-operator/config/samples/ray-cluster.separate-ingress.yaml`.
+# More information about change of setting was documented in https://github.com/ray-project/kuberay/pull/699 
+# and `ray-operator/config/samples/ray-cluster.separate-ingress.yaml`
+kubectl apply -f ray-operator/config/samples/ray-cluster.separate-ingress.yaml
+
+# Step 5: Check ingress created by Step 4.
```
Comment (DmitriGekhtman) [inline docs/guidance/ingress.md:188]:
```suggestion
# Step 5: Check the ingress created in Step 4.
```

**Comment by jasoonn** [comment]
@DmitriGekhtman Thanks for the suggestion. I have updated my pr.

**Review by DmitriGekhtman — APPROVED** [review]
Thanks!!

**Review by kevin85421 — APPROVED** [review]
LGTM. Let's wait for CI to finish. Thank @jasoonn for this contribution!

**Pull request merged**